peer.go

```diff
@@ -2387,62 +2391,64 @@ func (p *peer) WipeChannel(chanPoint *wire.OutPoint) error {
 // handleInitMsg handles the incoming init message which contains global and
 // local features vectors. If feature vectors are incompatible then disconnect.
 func (p *peer) handleInitMsg(msg *lnwire.Init) error {
-	p.remoteLocalFeatures = lnwire.NewFeatureVector(
-		msg.LocalFeatures, lnwire.LocalFeatures,
-	)
-	p.remoteGlobalFeatures = lnwire.NewFeatureVector(
-		msg.GlobalFeatures, lnwire.GlobalFeatures,
+	// First, merge any features from the legacy global features field into
+	// those presented in the local features fields.
+	err := msg.Features.Merge(msg.GlobalFeatures)
+	if err != nil {
```

Comment:
Just making sure this'll work: If an old node is sending overlapping global and local features (which is allowed), this will fail. I'm guessing we should be good since no such overlapping features (seemingly) was defined in the spec before. 

But the new spec specifies we should just take the "logical OR" of the vectors, which is not what we do here.

Reply:
it will only fail if they set optional and required for the same feature bit, which is a little stricter than the spec requires, but i think it's better to prevent unintended behavior. we still take the OR otherwise

Reply:
it will only fail if they set optional and required for the same feature bit, which is a little stricter than the spec requires, but i think it's better to prevent unintended behavior. we still take the OR otherwise

Reply:
hm, okay, I guess it couldn't hurt.